### DISCUSSION TITLE
Add Convention plugins

### DISCUSSION THREAD

**Discussion by krizzu** [AndroidDev-social/DodoForMastodon#86] [Ideas]
Hey,

With the project growing, a unified build setup would be priceless. I propose to create [convention plugins](https://docs.gradle.org/current/samples/sample_convention_plugins.html), so that all sub-modules (current and future) could benefit from it.

We could differentiate two plugins, one for code style (applying and configuring detect/spotless) and a build configuration (configuring compose/android min/compile version), but it's up for discussion.

I'm happy to start working on that.

**Comment by evant** [comment]
https://github.com/AndroidDev-social/DodoForMastodon/pull/103
